frontend/src/component/EmployeeForm.tsx

```diff
@@ -0,0 +1,214 @@
+import React, {ChangeEvent, FormEvent, useState} from "react";
+import {Employee, Address} from "../model/Employee";
+import {useNavigate} from "react-router-dom";
+import "../styling/AddEmployee.css"
+import moment from "moment";
+
+
+type Props = {
+    employee: Employee
+    onSubmit: (employee: Employee, file?: File) => void
+    action: "add" | "update"
+    navigateTo: string | undefined
+}
+
+export default function EmployeeForm(props: Props) {
+
+    const [firstName, setFirstName] = useState<string>(props.employee.firstName)
```

Comment:
I would suggest to implement `useReducer`. It can help to use a lot less code for the form :)
https://medium.com/swlh/usereducer-form-example-16675fa60229